composer/datasets/glue.py

```diff
@@ -33,21 +48,25 @@ class GLUEHparams(DatasetHparams):
     """Sets up a generic GLUE dataset loader.
 
     Args:
-        task (str): the GLUE task to train on, choose one from: CoLA, MNLI, MRPC, QNLI, QQP, RTE, SST-2, and STS-B.
-        tokenizer_name (str): The name of the HuggingFace tokenizer to preprocess text with.
-        split (str): Whether to use 'train', 'validation' or 'test' split.
-        max_seq_length (int): Optionally, the ability to set a custom sequence length for the training dataset.
-            Default: 256
-        num_workers (int): Optionally, the number of CPU workers to use to preprocess the text. Default: 64
-        max_network_retries (int): Optionally, the number of times to retry HTTP requests if they fail. Default: 10
+        task (str): the GLUE task to train on, choose one from: ``'CoLA'``, ``'MNLI'``,
```

Comment:
`Default: ``None``.`

Reply:
Does the "None" below mean "None" or a required argument? 
```
task: str = hp.optional(
         "The GLUE task to train on, choose one from: CoLA, MNLI, MRPC, QNLI, QQP, RTE, SST-2, and STS-B.", default=None)
 ```

Reply:
We should double check with @moinnadeem, but I think it's a required argument.

Reply:
It is a required argument, it has to be one of those, and will fail at validation otherwise!

Reply:
@moinnadeem's resposne: The `hp.optional` construction is necessary because Python dataclasses cannot have required arguments come after optional arguments. [GLUEHparams](https://github.com/mosaicml/composer/blob/dev/composer/datasets/glue.py#L48) inherits from [DatasetHparams](https://github.com/mosaicml/composer/blob/dev/composer/datasets/hparams.py#L51), which has optional arguments, so GLUEHparam arguments need to be optional as well.